### PR TITLE
Upgrade to Terraform 0.12+, other updates

### PR DESCRIPTION
**Pull request by lingrino**
- Update to new terraform 0.12 syntax
- Use Amazon Linux 2 over Centos
- Update vault version
- Remove a line break from the license
- Added more tagging to packer instances
- `alb_certificate_arn` in `main/variables.tf` was missing a default of `""`

I think we should tag this release after goes out as `2.0.0` and update the version more regularly in the future.

I tested everything manually and can confirm these changes work the same as before. We should really add CI. With some AWS access keys it shouldn't be too hard?

### CODE REVIEW

**Comment by lingrino** [comment]
@knehring Yooo who's supposed to review this now? Not many people have access that I can ping

**Comment by lingrino** [comment]
@knehring bump

**Comment by lingrino** [comment]
I am going to merge this at the end of today. There are issues asking for this support. Please review if you want.

File: packer/vault.json.example [hunk 1]
```diff
@@ -14,46 +14,58 @@
     {
       "name": "vault",
       "type": "amazon-ebs",
+      "communicator": "ssh",
+      "ssh_username": "ec2-user",
       "region": "{{ user `builder_region` }}",
       "source_ami_filter": {
         "filters": {
-          "name": "CentOS Linux 7 x86_64 HVM EBS *"
+          "name": "amzn2-ami-hvm-*-x86_64-gp2",
```
Comment (afpacket) [inline packer/vault.json.example]: Hey @lingrino - If you must, would you consider making the ami name and owner a variable?

File: packer/ansible/roles/vault/tasks/main.yml [hunk 2]
```diff
@@ -1,25 +1,18 @@
...
 - name: Update all packages
   package:
     name: '*'
     state: latest
 
-- name: Install EPEL
-  package:
-    name: "epel-release"
-    state: latest
-
```
Comment (afpacket) [inline packer/ansible/roles/vault/tasks/main.yml]: It looks like awscli is the only hard dependency here for needing EPEL, but in any case, another option is to execute this conditionally based on `ansible_distribution` like this:

```
- name: Install EPEL
  package:
    name: "epel-release"
    state: latest
  when: ansible_distribution in ["CentOS", "RedHat"]
```

File: packer/ansible/roles/vault/tasks/main.yml [hunk 3]
```diff
@@ -1,25 +1,18 @@
...
 - name: Install Necessary Packages
   package:
     name: "{{ item }}"
     state: latest
   with_items:
     - wget
     - unzip
-    - awscli
```
Comment (afpacket) [inline packer/ansible/roles/vault/tasks/main.yml]: In the spirit of multi-os, you could add something like this:

```
- name: install aws cli
  package:
    name: "{{ item }}"
    state: present
  with_items:
    - awscli
  when:
     - ansible_os_family == "RedHat"
     - ansible_system_vendor in ["Amazon EC2", "Xen"]
```

**Comment by lingrino** [comment]
@afpacket I definitely think there's value in supporting more than one OS. Check out the existing PR for that or maybe contribute something in a new PR after I merge this?

https://github.com/avantoss/vault-infra/pull/15

**Comment by lingrino** [comment]
I've pushed the `2.0.0` tag

**Pull request merged**